From a3df65edd98e6361e4f19751f24d68a8a629c0d8 Mon Sep 17 00:00:00 2001
From: "ola.lilja" <ola.lilja@intel.com>
Date: Fri, 27 Feb 2015 10:08:31 +0100
Subject: [PATCH] ASoC: Intel: Prevent illegal bp-int on ECS1/2

When mic is enabled/disabled it triggers button-press
interrupts. This is fixed by making sure we only allow
button-press when HS is active.

Issue: GMINL-6480

Change-Id: I5ea5c462fae863a261c458e1ed3c5ddad945cb43
Signed-off-by: ola.lilja <ola.lilja@intel.com>
---
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index 946ea87..4355d06 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -289,8 +289,9 @@ static int byt_micbias_interrupt(void *data)
 	if (cancel_delayed_work_sync(&drvdata->bp_recheck))
 		pr_debug("%s: bp-recheck interrupted!\n", __func__);
 
-	if (byt_bp_check(drvdata, false))
-		snd_soc_jack_report(jack, jack->status, SND_JACK_BTN_0);
+	if (jack->status & SND_JACK_MICROPHONE)
+		if (byt_bp_check(drvdata, false))
+			snd_soc_jack_report(jack, jack->status, SND_JACK_BTN_0);
 
 	mutex_unlock(&drvdata->jack_mlock);
 	return jack->status;
-- 
1.9.1

